rllib/policy/policy.py

```diff
@@ -621,11 +621,10 @@ def _initialize_loss_from_dummy_batch(
         sample_batch_size = max(self.batch_divisibility_req * 4, 32)
         self._dummy_batch = self._get_dummy_batch_from_view_requirements(
             sample_batch_size)
-        input_dict = self._lazy_tensor_dict(
-            {k: v
-             for k, v in self._dummy_batch.items()})
+        self._lazy_tensor_dict(self._dummy_batch)
         actions, state_outs, extra_outs = \
-            self.compute_actions_from_input_dict(input_dict, explore=False)
+            self.compute_actions_from_input_dict(
+                self._dummy_batch, explore=False)
         # Add all extra action outputs to view reqirements (these may be
         # filtered out later again, if not needed for postprocessing or loss).
         for key, value in extra_outs.items():
```

Comment:
I'd like to suggest another fix here:
self._dummy_batch[key] = np.zeros_like(value) Causes a problem when the values in the view requirements are a) non-zero, such as masks. All zero masks cause exceptions in Categorical Distributions as there are no possible sampling reults if they all all-zero. and b) if the data in the sample_batch is non-tensor such as a dict.

Reply:
My fix im my own branch for this is to just use self._dummy_batch[key] = value. I dont see that this could cause any issues as this is only a dummy batch. It also avoids assumptions around type/value of any custom extra_out values that are not part of the explicit view requirements.

Reply:
Good point! Thanks @Bam4d . I'll take a look and see whether this can be changed w/o breaking all tests :D

Reply:
^ Changed it. Let's wait for tests ...